File: js/src/maplestory/api/dto/character/characterPropensityDto.ts
import { CharacterPropensityDtoBody } from '../../response/character/characterPropensityDtoBody';

/**
 * 캐릭터 성향 정보
 */
class CharacterPropensityDto {
  date: Date | null;

  /**
   * 카리스마 레벨
   */
  charismaLevel: number | null;

  /**
   * 감성 레벨
   */
  sensibilityLevel: number | null;

  /**
   * 통찰력 레벨
   */
  insightLevel: number | null;

  /**
   *  의지 레벨
   */
  willingnessLevel: number | null;

  /**
   * 손재주 레벨
   */
  handicraftLevel: number | null;

  /**
   * 매력 레벨
   */
  charmLevel: number | null;

  constructor(obj: CharacterPropensityDtoBody) {
    const {
      date,
      charisma_level,
      sensibility_level,
      insight_level,
      willingness_level,
      handicraft_level,
      charm_level,
    } = obj;

    this.date = date ? new Date(date) : null;
    this.charismaLevel = charisma_level;
    this.sensibilityLevel = sensibility_level;
    this.insightLevel = insight_level;
    this.willingnessLevel = willingness_level;
    this.handicraftLevel = handicraft_level;
    this.charmLevel = charm_level;
  }
}

export { CharacterPropensityDto };

File: java/src/main/java/dev/spiralmoon/maplestory/api/tms/dto/character/CharacterPropensityDTO.java
package dev.spiralmoon.maplestory.api.tms.dto.character;

import com.fasterxml.jackson.annotation.JsonProperty;
import dev.spiralmoon.maplestory.api.Utils;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.time.ZonedDateTime;

/**
 * 角色性向資訊
 */
@AllArgsConstructor
@NoArgsConstructor
@Data
@ToString
public class CharacterPropensityDTO implements dev.spiralmoon.maplestory.api.common.dto.character.CharacterPropensityDTO {

    /**
     * 要搜尋的日期 (TST，每日資料中的小時與分鐘將顯示為 0)
     */
    @JsonProperty("date")
    private String date;

    /**
     * 領導力等級
     */
    @JsonProperty("charisma_level")
    private int charismaLevel;

    /**
     * 感性等級
     */
    @JsonProperty("sensibility_level")
    private int sensibilityLevel;

    /**
     * 洞察力等級
     */
    @JsonProperty("insight_level")
    private int insightLevel;

    /**
     * 意志等級
     */
    @JsonProperty("willingness_level")
    private int willingnessLevel;

    /**
     * 手藝等級
     */
    @JsonProperty("handicraft_level")
    private int handicraftLevel;

    /**
     * 魅力等級
     */
    @JsonProperty("charm_level")
    private int charmLevel;

    /**
     * 要搜尋的日期 (TST，每日資料中的小時與分鐘將顯示為 0)
     */
    public ZonedDateTime getDate() {
        return date != null
                ? Utils.toZonedDateTime(this.date)
                : null;
    }
}
